src/node.rs

```diff
@@ -729,6 +729,11 @@ impl Node {
             .collect()
     }
 
+    fn should_be_connected(&self, node: Name, blocks: &Blocks) -> bool {
+        let neighbours = nodes_in_any(blocks, &self.current_blocks);
+        neighbours.contains(&node)
```

Comment:
How do we handle newly joining nodes, which aren't yet a part of any block, but we should connect to them since they will be a part of our section? Is that somehow a different situation?